### PR TITLE
Include the domain name in the regular expression

### LINKED ISSUE #831: Bridge crashing with M_FORBIDDEN
**Describe the bug**
The bridge is crashing with M_FORBIDDEN error. I do not use t2bot.io so I don't know why this error is appearing.

**To Reproduce**
1. Start bot
2. Invite to room or try to send message
3. Open Logs

**Expected behavior**
The bot should not crash

**Logs**
```
Aug-12 18:08:06.817 [bot-sdkMatrixHttpClient] error: [
  '(REQ-2)',
  {
    errcode: 'M_FORBIDDEN',
    error: 'Application service has not registered this user (@_discord_472521620452278292:t2bot.io)'
  }
]

/node_modules/matrix-bot-sdk/src/http.ts:110
        throw new MatrixError(errBody, response.statusCode);
              ^
Error: M_FORBIDDEN: Application service has not registered this user (@_discord_472521620452278292:t2bot.io)
    at doHttpRequest (/node_modules/matrix-bot-sdk/src/http.ts:110:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at MatrixClient.descriptor.value (/node_modules/matrix-bot-sdk/src/metrics/decorators.ts:28:32)
    at MatrixClient.descriptor.value (/node_modules/matrix-bot-sdk/src/metrics/decorators.ts:28:32)
    at Intent.refreshJoinedRooms (/node_modules/matrix-bot-sdk/src/appservice/Intent.ts:296:33)
    at Intent.descriptor.value (/node_modules/matrix-bot-sdk/src/metrics/decorators.ts:83:32)
    at Appservice.processMembershipEvent (/node_modules/matrix-bot-sdk/src/appservice/Appservice.ts:616:9)
    at /node_modules/matrix-bot-sdk/src/appservice/Appservice.ts:841:21
```

**Additional context**
The bridge has been working fine for months before this error. Now it won't join rooms or sent from matrix to discord. Discord to matrix works fine though.

### PR DESCRIPTION
**Pull request by TheStranjer**
Resolves #831 

The generator will now add the domain name automatically.

* [x] I have added any new tests that need to pass to `sytest-whitelist` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/docs/CONTRIBUTING.md#sign-off)

Signed-off-by: Bob Arctor <neetzsche@tutanota.com>

### CODE REVIEW

**Review by Half-Shot — APPROVED** [review]
Change looks good but needs a changelog, see https://github.com/matrix-org/matrix-appservice-bridge/blob/develop/CONTRIBUTING.md#%EF%B8%8F-pull-requests

**Comment by TheStranjer** [comment]
@Half-Shot Try now.

File: changelog.d/834.bugfux [hunk 1]
```diff
@@ -0,0 +1 @@
+Include the domain name in the regular expression
```
Comment (Half-Shot) [inline changelog.d/834.bugfux]: 🫢 ITYM bugf*i*x

Comment (TheStranjer) [reply changelog.d/834.bugfux]: Lol funniest typo I've made in a while. Try now

**Pull request merged**